backend/routes/user.go

```diff
@@ -80,3 +81,22 @@ func getLastPlacedTime(w http.ResponseWriter, r *http.Request) {
 	// Return the last placed time in utc z format
 	WriteDataJson(w, "\""+string(lastTime.UTC().Format(time.RFC3339))+"\"")
 }
+
+
+func getUserColorVote(w http.ResponseWriter, r *http.Request) {
+	address := r.URL.Query().Get("address")
+	if address == "" {
+		WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	vote, err := core.PostgresQueryOne[int]("SELECT color_key FROM ColorVotes WHERE user_address = $1", address);
+
+	if err != nil {
+		WriteDataJson(w, "0")
```

Comment:
I know some parts of our code are doing this now, but going forward, it is better to use `COALESCE` instead of this, so errors can still be propagated.

So that would look like `SELECT COALESCE(color_key, 0) FROM ColorVotes WHERE user_address = $1`, so if there is no vote, 0 will be returned. 

Then here you can use the `WriteErrorJson` if there is an error